clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppController.java

```diff
@@ -103,8 +110,67 @@ public void run() {
                                 "Unable to parse inapp notification " + inAppNotification.getError());
                 return;
             }
-            inAppNotification.listener = inAppControllerWeakReference.get();
-            inAppNotification.prepareForDisplay(resourceProvider,templatesManager,storeRegistry);
+            prepareForDisplay(inAppNotification);
+        }
+
+        void prepareForDisplay(CTInAppNotification inApp) {
```

Comment:
Let us also change the class to not be a Runnable as it is not needed, also can we add tests in other PR?